Conflict 2.0

TYWong99 opened the pull request:
Reupload of `cli_sort_conflict.py` and `sort_conflict.py` to the latest branch, plus adding various property-fetching functions for `tiled_utils.py`.

The latest comment from the branch "Conflict 1.0" is also addressed here.

TYWong99 committed:
Conflict 2.0

Reupload of `cli_sort_conflict.py` and `sort_conflict.py` to the latest branch, plus adding various property-fetching functions for `tiled_utils.py`.

The latest comment from the branch "Conflict 1.0" is also addressed here.

qtran122 commented on logic/standalone/sort_conflict.py:
```diff
@@ -0,0 +1,190 @@
+def _GetSort( tiled_object ):
+    return tiled_utils.GetPropertyFromObject(tiled_object, '_sort')
+
+def GetParentName(obj, parent_map):
```
What is a 'parent'? Is it an object group layer? An XML element tree object? What is "obj"? Please write comments that tell what parent map is mapping to. "Maps X to Y"

TYWong99 replied:
I don't know how to name it either! It's just a map that I copied somewhere I can't recall.

I tried adding more to the description instead, LMK if that still confuses you by the time you see it.

Also it often doesn't matter whether something is treated as an object, objectgroup, XML tree object, or whatever applicable. They're probably handled in the same manner anyway, so there's little reason to distinguish them since it might over-complicate things IMO.

qtran122 commented on logic/standalone/sort_conflict.py:
```diff
@@ -0,0 +1,190 @@
+
+
+
+def PrintObjInfo(obj, p_name, max_obj_name_len = 0, max_layer_name_len = 0):
```
Bit of a misnomer, since this function doesn't do the actual "printing". How about "MakeInfoString".

What is obj? What is p_name? Let's rename these to be more descriptive. 

A quick summary of what all these variables do would be helpful! See this example from image_utils.py

![image](https://github.com/qtran122/StarTools/assets/126765399/2d7d7cee-dbac-4465-8cc7-72069e0edd34)

qtran122 commented on logic/standalone/sort_conflict.py:
```diff
@@ -0,0 +1,190 @@
+    return tiled_utils.GetPropertyFromObject(tiled_object, '_sort')
+
+def GetParentName(obj, parent_map):
+	return parent_map[obj].get('name')[8:]
```
Let's put a quick comment here to explain you're lobbing off the "objects_"

qtran122 commented on logic/standalone/sort_conflict.py:
```diff
@@ -0,0 +1,190 @@
+
+#--------------------------------------------------#
+
+# TODO Delete?
```
yes

qtran122 commented on logic/standalone/sort_conflict.py:
```diff
@@ -0,0 +1,190 @@
+#--------------------------------------------------#
+'''Utility'''
+
+def _GetSort( tiled_object ):
```
unused? I'm guessing the plan is to use this in the future. I'm okay with commenting it out, and then commenting it back in when it's ready to be used

qtran122 commented on logic/standalone/sort_conflict.py:
```diff
@@ -0,0 +1,190 @@
+
+
+def PrintPotentialConflicts(playdo, sortval_to_objects_map, list_scan_obj):
+	'''Print the dictionary that has been checked and sorted by their sort values'''
```
Let's give an expanded description. Like this function in image_utils

![image](https://github.com/qtran122/StarTools/assets/126765399/a5bfcfbb-7c23-46a4-b2da-1a5ebab7234d)

qtran122 commented on logic/common/tiled_utils.py:
```diff
@@ -220,6 +237,38 @@ def GetPolyPoints( input_string ):
     return poly_points
 
+# TODO overload function with same name, but accept both string or object?
+def GetPolyPointsFromObject( tiled_object ):
```
Python doesn't support function overloading they way you're used to (like in Jave and C#), since it's not strongly typed.

Now that there are 2 similarly named "GetPolyPoint" functions, I'd prefer more differentiation + more description. Let's change "GetPolyPoints" into "ConvertTiledPolyString". And let's expand the description to include example inputs & outputs

Similarly, let's expand GetPolyPointsFromObject. '''Accept object as input instead''' is not a proper description

qtran122 reviewed (changes requested):
Almost there. Mostly, I just want more descriptive comments and some name changes

TYWong99 committed:
Conflict 2.1

 - Rename `GetParentName()` into `GetParentObjectgroupName()`
 - Expanded descriptions for `PrintPotentialConflicts()` and `GetParentObjectgroupName()`
 - Relocated the (currently unused) `GetParentObject()` to `tiled_utils.py`
 - Deprecated the original `GetPolyPoints()` function, and replaced with one that accepts object directly
  - Adjust `aligner.py` used in `cli_water` accordingly
